### ISSUE TITLE
Standardize class/model/schema/instance terminology

### ISSUE THREAD

**Issue by reece** [ga4gh/vr-python#40]
The definitions of class, model, schema, instance, object, value object are related and overlap. The code is slightly confusing because it uses all of them in various places. Straighten this out to make it easier to understand. 

Definitions:
* schema: a conceptual representation of data and data relationships
* class: In code, a template or prototype of data and, optionally, methods. In a schema, a group of related data. 
* model: a _class_ whose primary purpose is to represent data
* object: any value in a language 
* instance: an object that is an exemplar of a class (or model)

In vr-python, let's use these terms:
* "schema" as above
* "class" for any schema class or python jsonschema class (that it calls models)
* "instance" as above

So, in other words, drop the "model" lingo.

**Comment by larrybabb** [comment]
I may be the outlier here, but i think the term "conceptual" which is used to define schema may be out of line the way we are using it. 

We are not defining a conceptual schema, we appear to be developing a physical schema which may or may not be based on a conceptual representation of concepts and their relationships in the genomics domain.

Our physical schema is the final result that is used by developers and implementers to build applications and share data (the end step in any useful specification). 

What we do not have is a conceptual form of the higher order concepts and their relationships that is in general agreement and drives this physical schema product.

If we had a conceptual form of the schema to then justify and rationalize the physical schema decisions that where made then I think there would be more transparency. But I will admit that every time we've attempted to go down that path it turns into a bunch of philosophical discussions and tends to cause challenges in moving forward.  

Nevertheless, I don't agree that our schema is a conceptual one. It is a physical one as far as I understand things. Again - I could be the outlier that needs to adjust my perspective. Glad to listen and continue working on methods for applying the physical schema to every day applications and enabling innovative uses.

**Comment by reece** [comment]
I agree with your use of "conceptual schema" and the distinction with "physical schema".  I also agree that I was too fast and loose with "schema" -- that can be conceptual or physical, and the thing we call a schema (the JSON schema) is indeed a physical model.

The flow of the design is Biological Definitions ⇒ Information Model ⇒ JSON Schema. The Information Model section is a conceptual model, right? That is, we do have a conceptual model, right?

**Comment by larrybabb** [comment]
I'm not sure our informational model = a conceptual model. Since it aligns 1:1 with the json schema it is simply a physical model.  I don't think we have a conceptual model. Whenever we attempt to create one we get caught up in a lot of discussion that is challenging to work through. However, we need "something" as a core conceptual model or we will continue to go back and forth on this (I think).
